### DISCUSSION TITLE
Missing required parameter when trying to deal with related resources

### DISCUSSION THREAD

**Discussion by linaspasv** [filamentphp/filament#6099] [Help]
I have ```MenuResource``` and ```CategoryResource```. Menu has many Categories and I would love to managed Categories related to Menu, so I have created a custom button in Menu List with a link generated to Category ``menus/{menu}/categories``

```php
  public static function table(Table $table): Table
  {
      return $table
          ->columns([
              Tables\Columns\TextColumn::make('title'),
          ])
          ->actions([

              Tables\Actions\Action::make('categories')
                  ->url(fn (Menu $record) => CategoryResource::getUrl('index', ['menu' => $record->id]))
                  ->icon('heroicon-o-tag')
                  ->color('success'),

              Tables\Actions\EditAction::make()

          ]);
  }
```

and in CategoryResource got a custom $slug like

```php
class CategoryResource extends Resource
{
    protected static ?string $model = Category::class;

    protected static ?string $slug = 'menus/{menu}/categories';

    protected static bool $shouldRegisterNavigation = false;

   .....
```

Custom link is generated correctly in the list of ```MenuResource``` but when I navigate through the link I get ```Missing required parameter``` error. I do understand that {menu} gets lost in action but is there a solution to what I am trying to achieve?

**Comment by Paulsky** [comment]
@linaspasv I had the same issue. Please see this discussion: https://github.com/filamentphp/filament/discussions/6127#discussion-5047397

I fixed it by adding this function in (in your case) `CategoryResource`

```
public static function getUrl(string $name = 'index', array $parameters = [], bool $isAbsolute = true, ?string $panel = null, ?Model $tenant = null): string
    {
        if (!isset($parameters['menu'])) {
            $parameters['menu'] = Route::current()->parameter('menu');
        }

        return parent::getUrl($name, $parameters, $isAbsolute, $panel, $tenant);
    }
```

You can read more about adding a nested resource here: https://laraveldaily.com/post/filament-nested-resources-courses-lessons **_(they use a different slug(=route path))_**

_**Edit**_: So, I think the problem is Livewire doesn't receive the parent resource parameter ('menu' in this case). The above method somewhat works. But it doens't when you use 'live' functions on the Form inputs, or use the (bulk) Delete Action. I think this is because, again, Livewire is not receiving the parent menu.

The only method I found to 'fix' all the '"parent resource" parameter not found' URL error's, is to check the previous URL, because if there is an AJAX request to for example /livewire/update, the current Route is /livewire/update (which doesn't include 'menu' ofcourse).

Please see these functions to ensure the 'parent resource' is set, and known, by Livewire:

```
public static function getUrl(string $name = 'index', array $parameters = [], bool $isAbsolute = true, ?string $panel = null, ?Model $tenant = null): string
    {
        if (!isset($parameters['menu'])) {
            $parameters['menu'] = self::getCurrentMenu(); //Route::current()->parameter('menu')
        }

        return parent::getUrl($name, $parameters, $isAbsolute, $panel, $tenant);
    }
```
```
private static function getCurrentMenu()
    {
        $menu = request('menu');
        if(isset($menu)){
            return $menu;
        }

        //try to get the Model from the previous known route
        $previousRoute = Route::getRoutes()->match(request()->create(url()->previousPath()));
        if (isset($previousRoute)) {
            //the model is not in the request (this is probably a Livewire request)
            //reset it
            $menu = $previousRoute->parameter('menu');
            request()->merge(['menu ' => $menu ]);
            return $menu;
        }

        return null;
    }
```
```
`public static function getEloquentQuery(): Builder
    {
        $menu = self::getCurrentMenu();
        if ($menu) {
            return parent::getEloquentQuery()->where('menu_id', $menu->id);
        }

        return parent::getEloquentQuery();
    }
```

I think this is very related to [https://github.com/filamentphp/filament/discussions/6127](https://github.com/filamentphp/filament/issues/7104)
